dataset/test/generated_test.cpp

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+#include <gtest/gtest.h>
+
+#include "scipp/dataset/data_array.h"
+#include "scipp/dataset/less.h"
+#include "scipp/dataset/reciprocal.h"
+#include "scipp/variable/less.h"
+#include "scipp/variable/reciprocal.h"
+
+#include "test_data_arrays.h"
+
+using namespace scipp;
+
+TEST(GeneratedUnaryTest, DataArray) {
+  const auto array = make_data_array_1d();
+  const auto out = reciprocal(array);
+  EXPECT_FALSE(out.data().is_same(array.data()));
+  EXPECT_EQ(out.data(), reciprocal(array.data()));
+  EXPECT_EQ(out.coords(), array.coords());
+  EXPECT_EQ(out.masks(), array.masks());
+  EXPECT_EQ(out.attrs(), array.attrs());
+  // Meta data is shallow-copied but dicts are not shared
+  EXPECT_NE(&out.coords(), &array.coords());
+  EXPECT_NE(&out.masks(), &array.masks());
+  EXPECT_NE(&out.attrs(), &array.attrs());
+  EXPECT_TRUE(out.coords()[Dim::X].is_same(array.coords()[Dim::X]));
+  EXPECT_TRUE(out.masks()["mask"].is_same(array.masks()["mask"]));
+  EXPECT_TRUE(out.attrs()[Dim("attr")].is_same(array.attrs()[Dim("attr")]));
```

Comment:
I only see one line that differs from `check_meta`. Can you reuse that function or annotate the difference(s)?

Reply:
Hah, and that single line difference is actually masks that should also get deep-copied for unaries!

Reply:
Fixed!